custom_components/tahoma/cover.py

```diff
@@ -105,14 +105,17 @@ def current_cover_position(self):
 
         None is unknown, 0 is closed, 100 is fully open.
         """
-
         position = self.select_state(
             CORE_CLOSURE_STATE,
             CORE_DEPLOYMENT_STATE,
             CORE_PEDESTRIAN_POSITION_STATE,
             CORE_TARGET_CLOSURE_STATE,
         )
 
+        # Uno devices can have a position not in 0 to 100 range when unknown
+        if position is None or position < 0 or position > 100:
+            return None
+
         if position is not None and "Horizontal" not in self.device.widget:
```

Comment:
This check `if position is not None` is now useless.

Reply:
Good catch!